design/design-doc-aggregate-resources.md

```diff
@@ -0,0 +1,965 @@
+  # managed resources; they respect the reclaimPolicy of the individual resource
+  # classes used to configure them.
+  reclaimPolicy: Delete
+  # AggregateResources dynamically provisioned using this AggregateResourceClass
+  # will write their connection secrets to this namespace.
+  writeConnectionSecretsToNamespace: crossplane-system
+---
+apiVersion: azure.crossplane.io/v1alpha3
+kind: ResourceGroupClass
+metadata:
+  name: private-mysql-server
+  labels:
+    # This ResourceGroupClass aggregates to the private-mysql-server
+    # AggregateResourceClass.
+    resourceclass.crossplane.io/aggregate-to: private-mysql-server
+specTemplate:
+  location: Central US
+  providerRef:
+    name: example
+---
+apiVersion: database.azure.crossplane.io/v1beta1
+kind: SQLServerClass
+metadata:
+  name: private-mysql-server
+  labels:
+    resourceclass.crossplane.io/aggregate-to: private-mysql-server
+specTemplate:
+  forProvider:
+    administratorLogin: my-cool-login
+    # MySQLServer managed resources dynamically provisioned using this resource
+    # class will select the ResourceGroup dynamically provisioned by the above
+    # resource class because its classRef will be named private-mysql-server and
+    # its aggregateRef will be identical to the MySQLServer's.
+    resourceGroupNameSelector:
+      matchAggregate: true
+      matchClassRef:
+        name: private-mysql-server
+    location: Central US
+    sslEnforcement: Disabled
+    version: "5.6"
+    sku:
+      tier: GeneralPurpose
+      capacity: 2
+      family: Gen5
+    storageProfile:
+      storageMB: 25600
+      backupRetentionDays: 7
+      geoRedundantBackup: Disabled
+  providerRef:
+    name: example
+  # MySQLServer managed resources dynamically provisioned using this class still
+  # write their own unaggregated, unadulterated connection secrets.
+  writeConnectionSecretsToNamespace: crossplane-system
+---
+apiVersion: database.azure.crossplane.io/v1alpha3
+kind: MySQLServerVirtualNetworkRuleClass
+metadata:
+  name: private-mysql-server
+  labels:
+    resourceclass.crossplane.io/aggregate-to: private-mysql-server
+spec:
+  serverNameSelector:
+    matchAggregate: true
+    matchClassRef:
+      name: private-mysql-server
+  resourceGroupNameSelector:
+    matchAggregate: true
+    matchClassRef:
+      name: private-mysql-server
+  properties:
+    # Aggregated resource classes can still reference a statically provisioned
+    # managed resource as they can today. Every MySQLVirtualNetworkRule created
+    # using this MySQLServerVirtualNetworkClass will allow traffic from the
+    # 'private' Subnet.
+    virtualNetworkSubnetIdRef:
+      name: private
+  providerRef:
+    name: example
+```
+
+Note that the `AggregateResourceClass` was annotated as the default, and will
+thus be used by any resource claim that does not specify a `resourceRef`,
+`classRef`, or `classSelector`. The resource claim API is unchanged under this
```

Comment:
I believe so - I think I'll try clarify this much earlier in the doc though. Several comments on the doc at the moment assume we'd support an arbitrary binding of claim to managed resource. This is not the case, but it's easy to miss because it's not explained until down in the technical implementation section.

Reply:
I pushed that change, so will resolve this.